buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchCluster.java

```diff
@@ -124,12 +133,21 @@ public void setNumberOfNodes(int numberOfNodes) {
             );
         }
 
+        if (numberOfNodes < zoneCount) {
```

Comment:
This logic looks flawed a bit to me: 
 - we always add one node on initialization [1]
 - as such, the number of nodes should be `+1`ed (or zone count `-1`ed): `if ((numberOfNodes + 1) < zoneCount)`
 - `currentZone = (currentZone >= zoneCount) ? 1 : (currentZone + 1);` would lead to uneven distribution of nodes (when there are more nodes than zones),  we probably should use  `currentZone = i % zoneCount + 1` assignment

[1] https://github.com/opensearch-project/OpenSearch/pull/3958/files#diff-a22e2ce210c503bbd47be71cc6e770064e8a9e008738d28fee5b381918f1fb8cR109

Reply:
Yes. We can see uneven distribution of nodes if ```numNodes/numZones != 0```. So if we have 3 nodes we need numZones to be 1 or 3. So that all zones has equal nodes or one zone has all nodes. 

- We can add validation to make sure if numNodes can be evenly added in zones. Let me know if this seems required. I felt to keep it for developer working on local to configure as per his choice.

Reply:
> Yes. We can see uneven distribution of nodes if numNodes/numZones != 0

This is fine we just should not fill single zone

> We can add validation to make sure if numNodes can be evenly added in zones.

I don't think it is needed, doing best effort assignment is sufficient

Reply:
Ok.. Tested we assign zones as per the node count. Working as expected.

Reply:
TY

Reply:
> Ok.. Tested we assign zones as per the node count. Working as expected. https://github.com/opensearch-project/OpenSearch/pull/3958/files#diff-a22e2ce210c503bbd47be71cc6e770064e8a9e008738d28fee5b381918f1fb8cR151

I think this is still not right: `return currentZone >= zoneCount ? 1 : (currentZone + 1);` (see please https://github.com/opensearch-project/OpenSearch/pull/3958/files#r929143271)

Reply:
Updated to use different branch for zone and without zone code path. Also using now mod operation to decide the zone number.